src/MutableVertexPartition.cpp

```diff
@@ -777,7 +812,6 @@ void MutableVertexPartition::cache_neigh_communities(size_t v, igraph_neimode_t
 
   // Reset cached neighbours
   _cached_neighs_comms->clear();
-  _cached_neighs_comms->reserve(degree);
```

Comment:
Any particular reason why you removed this reservation? It doesn't matter much, but I would think that making sure that all neighbour communities are guaranteed to fit in the vector would be preferable.

Reply:
I just wanted to avoid repeatedly calling `reserve()`.

I moved this reservation to `MutableVertexPartition::init_admin` since the number of neighboring communities should never exceed the number of nodes in the graph.
https://github.com/vtraag/leidenalg/blob/55849af4b0b1b4b54fdd76cd948e52651c948205/src/MutableVertexPartition.cpp#L151-L156
If I recall correctly, this was a pretty minor performance improvement.

Regardless, this function only calls `push_back` on the vector, which would reallocate if necessary.